tomcat 6x-8x integration

Once running (on https://github.com/p6spy/p6spy-it):

    export ORG_GRADLE_PROJECT_container=tomcat6x; ./gradlew clean unitTest 

I get Error:

		    Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: private com.p6spy.engine.spy.impl.ContactRepository com.p6spy.engine.spy.impl.CrossRepoService.contactRepository; nested exception is
	 org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [ApplicationContext.Driver.xml]: Invocation of init method failed; nested exception is liquibase.exc
	eption.DatabaseException: org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (Unable to find a driver that accepts jdbc:h2:tcp://localhost:9091//home/pb/all/prg/oss/p6spy-it/build/h2/s1/test3)
		at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:508)
		at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:87)
		at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:289)
		... 46 more
	    Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [ApplicationContext.Driver.xml]: Invocation of init method failed; nested exception is
	 liquibase.exception.DatabaseException: org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (Unable to find a driver that accepts jdbc:h2:tcp://localhost:9091//home/pb/all/prg/oss/p6spy-it/build/
	h2/s1/test3)
		at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1553)
		at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:539)
		at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:475)
		at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:304)
		at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:228)
		at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:300)
		at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:195)
		at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:294)
		at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:195)
		at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1017)
		at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:960)
		at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:858)
		at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:480)
		... 48 more
	    Caused by: liquibase.exception.DatabaseException: org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (Unable to find a driver that accepts jdbc:h2:tcp://localhost:9091//home/pb/all/prg/oss/p
	6spy-it/build/h2/s1/test3)
		at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:293)
		at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1612)
		at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1549)
		... 60 more
	    Caused by: org.apache.tomcat.dbcp.dbcp.SQLNestedException: Cannot create PoolableConnectionFactory (Unable to find a driver that accepts jdbc:h2:tcp://localhost:9091//home/pb/all/prg/oss/p6spy-it/build/h2/s1/test3)
		at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1549)
		at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createDataSource(BasicDataSource.java:1388)
		at org.apache.tomcat.dbcp.dbcp.BasicDataSource.getConnection(BasicDataSource.java:1044)
		at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:288)
		... 62 more
	    Caused by: java.sql.SQLException: Unable to find a driver that accepts jdbc:h2:tcp://localhost:9091//home/pb/all/prg/oss/p6spy-it/build/h2/s1/test3
		at com.p6spy.engine.spy.P6SpyDriver.findPassthru(P6SpyDriver.java:115)
		at com.p6spy.engine.spy.P6SpyDriver.connect(P6SpyDriver.java:90)
		at org.apache.tomcat.dbcp.dbcp.DriverConnectionFactory.createConnection(DriverConnectionFactory.java:38)
		at org.apache.tomcat.dbcp.dbcp.PoolableConnectionFactory.makeObject(PoolableConnectionFactory.java:582)
		at org.apache.tomcat.dbcp.dbcp.BasicDataSource.validateConnectionFactory(BasicDataSource.java:1556)
		at org.apache.tomcat.dbcp.dbcp.BasicDataSource.createPoolableConnectionFactory(BasicDataSource.java:1545)
		... 65 more


as described in our docs (http://p6spy.github.io/p6spy/2.0/install.html#tomcat), tomcat requires section similar to:

     driverlist=com.mysql.jdbc.Driver

I'm not sure about the cause, but do believe, that eager initialization of p6spy is missing here (https://github.com/p6spy/p6spy/commit/f4afcb362223af8c69aff75033e5c1787036b54b#diff-4b14eb1503c82de0bcf667723ff92964L59). There used to be in the `P6DataSource.java` following:

    static {
      // make sure that the core has been initialized
      P6Core.initialize();
    }

Do you guys have any idea how to proceed here?

I've tried adding some eager stuff into `P6DataSource.java`, namely:

      static {
		P6ModuleManager.getInstance();
	}

but that didn't help, so I'm starting to doubt I'm looking at a rigfht cause here.

--- a/src/main/java/com/p6spy/engine/spy/P6SpyDriver.java
+++ b/src/main/java/com/p6spy/engine/spy/P6SpyDriver.java
@@ -42,9 +42,6 @@ public class P6SpyDriver implements Driver {
   static {
     try {
       DriverManager.registerDriver(INSTANCE);
-      
-      // registers the passthru drivers, if configured s
-  	  P6ModuleManager.getInstance();
     } catch (SQLException e) {
       throw new IllegalStateException("Could not register P6SpyDriver with DriverManager", e);
     }
@@ -103,6 +100,9 @@ public class P6SpyDriver implements Driver {
   }
 
   protected Driver findPassthru(String url) throws SQLException {
+    // registers the passthru drivers, if configured s
+    P6ModuleManager.getInstance();
+    
     String realUrl = extractRealUrl(url);
     Driver passthru = null;
     for (Driver driver: registeredDrivers() ) {

--- a/src/main/java/com/p6spy/engine/spy/P6SpyDriver.java
+++ b/src/main/java/com/p6spy/engine/spy/P6SpyDriver.java
@@ -42,6 +42,9 @@ public class P6SpyDriver implements Driver {
   static {
     try {
       DriverManager.registerDriver(INSTANCE);
+      
+      // registers the passthru drivers, if configured s
+  	  P6ModuleManager.getInstance();
     } catch (SQLException e) {
       throw new IllegalStateException("Could not register P6SpyDriver with DriverManager", e);
     }

